wp_bak/app/wp_bak.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+# Docker:
+# docker build -t colindavey/wp_bak:alpha .
+# docker run -v `pwd`/src:/home/mvp/app/src -v `pwd`/dst:/home/mvp/app/dst -t colindavey/wp_bak:alpha
+
+from datetime import datetime
+from fnmatch import fnmatch
+import logging
+import os
+import shutil
+import sys
+import tarfile
+import time
+
+#############################
+# CONSTANTS
+#############################
+
+SRC_DIR='src/'
+DST_DIR='dst/'
+SHORT_DIR=DST_DIR+'shorts/'
+LONG_DIR=DST_DIR+'longs/'
+
+MINUTE = 60
+HOUR = 60 * MINUTE
+DAY = 24 * HOUR
+WEEK = 7 * DAY
+
+DATE_TIME_FORMAT = "%Y-%m-%d-%H-%M-%S"
+
+# Duration between backups in seconds
+# Actual value
+# SHORT_DURATION = DAY
+# Value for testing
+SHORT_DURATION = 1
+# Delete everything in the shorts folder older than this
+SHORT_DURATION_TO_KEEP = 7 * SHORT_DURATION
+
+LONG_DURATION = 30 * SHORT_DURATION
+# Delete everything in the longs folder older than this
+LONG_DURATION_TO_KEEP = 2 * LONG_DURATION
+
+# Set to a value for testing
+# NUM_REPS = None
+NUM_REPS = 100
+
+logging.basicConfig(level=logging.INFO, stream=sys.stderr)
+log = logging.getLogger(__name__)
+
+#############################
+# FUNCTIONS
+#############################
+
+def mkdir_if_not_exist(dir):
+    """Makes a directory if it doesn't exist.
+
+    Parameters
+    ----------
+    dir : string, name of directory
+        The directory to make.
+    """
+    if not os.path.exists(dir):
+        os.makedirs(dir)
+
+def filename2age(now, filename):
+    """Calculcates the age of a file, in seconds from the files name.
+
+    Parameters
+    ----------
+    now : datetime object
+        The time to use as "now" for calculating item's age.
+
+    filename : string, path of file
+        The name of file to calculate age of.
+    """
+    # Get string representation of archive date
+    date = filename.replace('archive', '').replace('.tar.gz', '')
+    # Get age from archive-date string
+    age = now - datetime.strptime(date, DATE_TIME_FORMAT)
+    return (age.days * DAY) + age.seconds
```

Comment:
`age` is a `timedelta` which is easy to use so why not just return that. `timdelta` support comparisons like `>` and `<`, etc.

Reply:
Done.